lib/optimizer.py

```diff
@@ -88,24 +88,55 @@ def _WorksBetterOnSomeOtherClip(self, encoding, bitrate, videofile):
           return best_on_this
     return None
 
-  def _EncodingWithOneLessParameter(self, encoding, bitrate, videofile):
+  def _EncodingWithOneLessParameter(self, encoding, bitrate, videofile,
+                                    considered_seen):
     """Find an untried encoder that has one parameter less than this one."""
+    # pylint: disable=R0201
     new_encoder = encoding.encoder.RandomlyRemoveParameter()
     if not new_encoder:
       return None
+    if considered_seen and new_encoder.Hashname() in considered_seen:
```

Comment:
A better way to go about this is to add this to the beginning:

considered_seen = considered_seen or set() and just check if the hashname is in considered_seen.